sdk/device.go

```diff
@@ -398,11 +407,15 @@ func (device *Device) encode() *synse.Device {
 // If duplicate IDs are detected, the plugin will terminate.
 func updateDeviceMap(devices []*Device) {
 	for _, d := range devices {
-		if _, hasDevice := ctx.devices[d.GUID()]; hasDevice {
+		if existing, hasDevice := ctx.devices[d.GUID()]; hasDevice {
 			// If we have devices with the same ID, there is something very wrong
 			// happening and we will not want to proceed, since we won't be able
 			// to route to devices correctly.
-			log.Fatalf("[sdk] duplicate device id found: %s", d.GUID())
+			deviceJSON, err := existing.JSON()
+			if err != nil {
+				log.Fatalf("[sdk] duplicate device (failed to dump JSON: %v): %v", err, existing)
+			}
+			log.Fatalf("[sdk] duplicate id found for device: %v", deviceJSON)
```

Comment:
Please dump the devices / fields that are duplicates. There are two of them. (deviceJSON is duplicate of what?)